### PR TITLE
gnome-hexgl: init at unstable-2019-08-21

### PR DESCRIPTION
**Pull request by jtojnar**

### CODE REVIEW

File: pkgs/development/libraries/gthree/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    graphene
+    json-glib
+  ];
+
+  mesonFlags = [
+    "-Dgtk_doc=true"
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Gthree port of HexGL";
```
Comment (worldofpeace) [inline pkgs/development/libraries/gthree/default.nix]:
```suggestion
    description = "GObject/Gtk port of three.js ";
```

or

```suggestion
    description = "Port of three.js to GObject and Gtk3";
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test grahpene

File: pkgs/development/libraries/gthree/default.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  propagatedBuildInputs = [
+    glib
+    gtk3
+    graphene
+    json-glib
+  ];
+
+  mesonFlags = [
+    "-Dgtk_doc=true"
```
Comment (worldofpeace) [inline pkgs/development/libraries/gthree/default.nix]: From looking at ofborg darwin has problems with the documentation, and it fails to build.

Comment (jtojnar) [reply pkgs/development/libraries/gthree/default.nix]: Must be another instance of https://github.com/mesonbuild/meson/issues/5202

File: pkgs/development/libraries/mutest/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+stdenv.mkDerivation rec {
+  name = "mutest";
+  version = "unstable-2019-10-12";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "ebassi";
+    repo = "mutest";
```
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/development/libraries/mutest/default.nix]: @timokau nicely described my feelings on this here: https://github.com/NixOS/nixpkgs/pull/65480#discussion_r315799884

[the remark this answers is not in the document]
Comment (worldofpeace) [reply pkgs/development/libraries/mutest/default.nix]: Probably should have thought of that when I wrote the pantheon expressions. Would have saved me #66136 :rofl:

**Comment by worldofpeace** [comment]
@jtojnar It looks like the graphene nixos test fails now.

**Comment by jtojnar** [comment]
The NixOS tests are annoying like that, I need to redirect `stderr` to `stdout` like `gnome-desktop-testing-runner -d '${pkgs.graphene.installedTests}/share' graphene-1.0/quad.test 2>&1` to be able to see everything in context:

```
Running test: graphene-1.0/quad.test

  graphene_quad_t
    has bounds
      ✓ converting rect corners into a quad yields the same bounds
      ✓ converting a rect into a quad yields the same bounds
      ✓ point[0] to be top left
free(): invalid pointer
FAIL: graphene-1.0/quad.test (Child process killed by signal 6)
```

Opened an issue: https://github.com/ebassi/graphene/issues/168

**Comment by worldofpeace** [comment]
Yeah that test passes in my user environment as well.

**Review by worldofpeace — APPROVED** [review]
LGTM.

**Comment by jtojnar** [comment]
Maybe some QEMU SIMD interaction?

**Comment by worldofpeace** [comment]
> Maybe some QEMU SIMD interaction?

Possibly, though my knowledge in that area is limited.

The only change that happened to the test between the releases was
* https://github.com/ebassi/graphene/commit/62f50c57991651d38d288761e8855808a61b59cd#diff-2f020a9bb1dfd904a4a0c5b7fc73ce38

**Pull request merged**